### ISSUE TITLE
Release as a framework

### ISSUE THREAD

**Issue by ethanresnick** [TechAtNYU/intranet#105]
Just a v0 w/ no compatibility promises

**Comment by maxdumas** [comment]
sure np bro

**Comment by freialobo** [comment]
^thats so cute

**Comment by ethanresnick** [comment]
@maxdumas sarcasm?

**Comment by maxdumas** [comment]
@ethanresnick Yeah haha, as is the small tag. I realized we didn't talk about this at all at the meeting, although this is a pretty serious consideration that pretty much everything I do in the intranet going forward should be mindful of. I'm still not totally sure how to approach this; I'm not sure about the specifics of library design on top of Angular. That said I am going to put a heavy emphasis going forward on all specific values being provided via configuration or data.

**Comment by ethanresnick** [comment]
Using more configuration options sounds like a good start. And, yeah, the small tag should've been a tipoff. That said, are we really that far from being able to release this? I mean, most of the code is already generic/decoupled from any tech@nyu-specific purpose, right? (Except for the logic stuff.)

**Comment by maxdumas** [comment]
Right. It's more of an issue of what to decouple, and how it should be decoupled. A lot of our logic is in the controllers right now which don't really make sense to put into some framework/library; as far as I understand pretty much anything that would go into an Angular framework should be a service. To prepare for that I've started moving some of our stuff to services.

**Comment by AbhiAgarwal** [comment]
Also! comments + documentation

**Comment by ethanresnick** [comment]
> Right. It's more of an issue of what to decouple, and how it should be decoupled.

Totally fair. I was trying to address this originally by just saying "let's release it as a v0", which, under semver, means we can break whatever we want and no one can complain. So we could refactor as we go. But the impetus for that was primarily to launch simultaneously with JSON API. Now, since we've already missed that deadline, we can spend probably spend a little more time getting the architecture right before we put out anything meant for public use. That'll save us time on docs etc in the long run, which, as Abhi points out, would be another to do.
